File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/config/WanService.java
package com.yyxnb.module_wanandroid.config;

import com.yyxnb.module_wanandroid.bean.WanAriticleBean;
import com.yyxnb.module_wanandroid.bean.WanClassifyBean;
import com.yyxnb.common_res.bean.WanData;
import com.yyxnb.module_wanandroid.bean.WanNavigationBean;
import com.yyxnb.module_wanandroid.bean.WanStatus;
import com.yyxnb.module_wanandroid.bean.WanSystemBean;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

import static com.yyxnb.common_res.config.BaseAPI.HEADER_WAN;

public interface WanService {

    /**
     * banner
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("banner/json")
    Observable<WanData<List<WanAriticleBean>>> getBanner();

    /**
     * 首页数据
     *
     * @param page
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("article/list/{page}/json")
    Observable<WanData<WanStatus<WanAriticleBean>>> getAritrilList(@Path("page") int page);

    /**
     * 置顶数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("article/top/json")
    Observable<WanData<List<WanAriticleBean>>> getTopAritrilList();

    /**
     * 项目分类
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("project/tree/json")
    Observable<WanData<List<WanClassifyBean>>> getProjecTypes();

    /**
     * 根据分类id获取项目数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("project/list/{page}/json")
    Observable<WanData<WanStatus<WanAriticleBean>>> getProjecDataByType(@Path("page") int page, @Query("cid") int cid);

    /**
     * 获取最新项目数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("article/listproject/{page}/json")
    Observable<WanData<List<WanAriticleBean>>> getProjecNewData(@Path("page") int page);

    /**
     * 获取热门搜索数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("hotkey/json")
    Observable<WanData<List<WanClassifyBean>>> getSearchData();

    /**
     * 根据关键词搜索数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @POST("article/query/{page}/json")
    Observable<WanData<WanStatus<WanAriticleBean>>> getSearchDataByKey(@Path("page") int page, @Query("k") String searchKey);

    /**
     * 公众号分类
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("wxarticle/chapters/json")
    Observable<WanData<List<WanClassifyBean>>> getPublicTypes();

    /**
     * 获取公众号数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("wxarticle/list/{id}/{page}/json")
    Observable<WanData<WanStatus<WanAriticleBean>>> getPublicData(@Path("page") int page, @Path("id") int id);

    /**
     * 获取体系数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("tree/json")
    Observable<WanData<List<WanSystemBean>>> getSystemData();

    /**
     * 知识体系下的文章数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("article/list/{page}/json")
    Observable<WanData<List<WanAriticleBean>>> getAritrilDataByTree(@Path("page") int page, @Query("cid") int cid);

    /**
     * 广场列表数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("user_article/list/{page}/json")
    Observable<WanData<WanStatus<WanAriticleBean>>> getSquareData(@Path("page") int page);

    /**
     * 获取导航数据
     *
     * @return
     */
    @Headers(HEADER_WAN)
    @GET("navi/json")
    Observable<WanData<List<WanNavigationBean>>> getNavigationData();


}

File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/config/WanService.kt
package com.yyxnb.module_wanandroid.config

import com.yyxnb.common_base.config.BaseAPI
import com.yyxnb.module_wanandroid.bean.*
import retrofit2.http.*

interface WanService {
    /**
     * banner
     *
     * @return
     */
    @GET("banner/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun banner(): WanData<List<WanAriticleBean>>

    /**
     * 首页数据
     *
     * @param page
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("article/list/{page}/json")
    suspend fun getAritrilList(@Path("page") page: Int): WanData<WanStatus<WanAriticleBean>>

    /**
     * 置顶数据
     *
     * @return
     */
    @GET("article/top/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun topAritrilList(): WanData<List<WanAriticleBean>>

    /**
     * 项目分类
     *
     * @return
     */
    @GET("project/tree/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun projecTypes(): WanData<List<WanClassifyBean>>

    /**
     * 根据分类id获取项目数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("project/list/{page}/json")
    suspend fun getProjecDataByType(@Path("page") page: Int, @Query("cid") cid: Int): WanData<WanStatus<WanAriticleBean>>

    /**
     * 获取最新项目数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("article/listproject/{page}/json")
    suspend fun getProjecNewData(@Path("page") page: Int): WanData<List<WanAriticleBean>>

    /**
     * 获取热门搜索数据
     *
     * @return
     */
    @GET("hotkey/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun searchData(): WanData<List<WanClassifyBean>>

    /**
     * 根据关键词搜索数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @POST("article/query/{page}/json")
    suspend fun getSearchDataByKey(@Path("page") page: Int, @Query("k") searchKey: String?): WanData<WanStatus<WanAriticleBean>>

    /**
     * 公众号分类
     *
     * @return
     */
    @GET("wxarticle/chapters/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun publicTypes(): WanData<List<WanClassifyBean>>

    /**
     * 获取公众号数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("wxarticle/list/{id}/{page}/json")
    suspend fun getPublicData(@Path("page") page: Int, @Path("id") id: Int): WanData<WanStatus<WanAriticleBean>>

    /**
     * 获取体系数据
     *
     * @return
     */
    @GET("tree/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun systemData(): WanData<List<WanSystemBean>>

    /**
     * 知识体系下的文章数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("article/list/{page}/json")
    suspend fun getAritrilDataByTree(@Path("page") page: Int, @Query("cid") cid: Int): WanData<List<WanAriticleBean>>

    /**
     * 广场列表数据
     *
     * @return
     */
    @Headers(BaseAPI.HEADER_WAN)
    @GET("user_article/list/{page}/json")
    suspend fun getSquareData(@Path("page") page: Int): WanData<WanStatus<WanAriticleBean>>

    /**
     * 获取导航数据
     *
     * @return
     */
    @GET("navi/json")
    @Headers(BaseAPI.HEADER_WAN)
    suspend fun navigationData(): WanData<List<WanNavigationBean>>
}